### DISCUSSION TITLE
Melhor forma para retornar statuscode em casos de erro

### DISCUSSION THREAD

**Discussion by RonaldoBueno16** [balta-io/forum#404] [Tira Dúvidas] [answered]
Boa tarde!

Eu estou aplicando os conhecimentos que aprendi no curso, mas em outro cenário!
Agora para um sistema para adicionar o email na fila, porém o e-mail deve estar no formato correto!

Para isso, tenho a validação do contrato Email no Flunt:
`public void Validate()
    {
        AddNotifications(
            new Contract()
                .Requires()
                .IsEmail(From, "From", "Digite um e-mail válido")
                .IsEmail(To, "To", "Digite um e-mail válido")
        );
    }`

E no meu manipulador do comando, eu verifico se é invalido, se for, retorno um resultado genérico.
`public ICommandResult Handle(AddMailQueueCommand queueCommand)
    {
        queueCommand.Validate();
        if (queueCommand.Invalid)
            return new GenericCommandResult(false, "Ops, parece que o seu email esta inválido!",
                queueCommand.Notifications);

        try
        {
            _service.Enqueue(queueCommand);
            return new GenericCommandResult(true, "Email adicionado a fila", null);
        }
        catch (Exception error)
        {
            return new GenericCommandResult(false, "Ocorreu um erro inesperado ao enviar o e-mail para a fila", error);
        }
    }`

Porém eu não sei como eu posso retornar o status code 400 caso a requisição esteja errada (queueCommand.Invalid) e também quando cai na excessão e ocorre algum erro no servidor.

Eu tenho o meu controller: 
`using Microsoft.AspNetCore.Mvc;
using SendMail.Domain.Commands;
using SendMail.Domain.Handlers;
using SendMail.Domain.Services;

namespace SendMail.Api.Controllers;

[ApiController]
[Route("v1")]
public class MailController : ControllerBase
{
    [Route("sendmail")]
    [HttpPost]
    public GenericCommandResult Send(
        [FromBody] AddMailQueueCommand command,
        [FromServices] MailQueueHandler handler)
    {
        var result = (GenericCommandResult)handler.Handle(command);
        return result;
    }
}`

Qual a melhor forma de eu retornar esse erro ao usuário?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @RonaldoBueno16, como vai?

Poderia ser algo assim:
```csharp
if (result.IsSuccess)
        return StatusCode(400, OBJETO DE ERRO)

return Ok(objeto de sucesso)
```
